### DISCUSSION TITLE
Not able to patch an SPO list item column of type URL field

### DISCUSSION THREAD

**Discussion by ctxlabeu** [microsoftgraph/msgraph-sdk-go#687] [Q&A] [answered]
Dear community,

I've recently started to adopt Go and MS Graph Go SDK to replace PowerShell solutions.

I am struggling to update an SPO list item with the graph-go-sdk, respectively the graph-go-beta-sdk.
I do can update the list item when leaving the Url column out of the request, but as soon as I add it, I am getting a "**_Invalid request_**" response from MS Graph respectively the graph client.

This is the list column definition:
<img width="576" alt="image" src="https://github.com/microsoftgraph/msgraph-sdk-go/assets/12398189/9e44703f-9ecc-4aa6-9d90-79c2f3930a38">

Updating the items without the column "PortalLink" works (string, date/time fields) - if I want to also patch it with the request, it fails.

See the code snippet.
I've tried to define that specific column as []string, map[string]string & map[string]interface{}. Non works....

```go
func (g *GraphServiceClient) UpdateListItem(siteId string, listId string, newAccessCode string, portal portal.WifiPortal) (models.FieldValueSetable, error) {

	requestBody := models.NewFieldValueSet()
	itemTitle := "GUEST WIFI access code " + portal.PortalName
	itemUrl := map[string]interface{}{
		"Description": "https://graph.microsoft.com",
		"Url":         "https://graph.microsoft.com",
	}
	additionalData := map[string]interface{}{
		"Title":      itemTitle,
		"Accesscode": newAccessCode,
		"Validsince": time.Now(),
		"Region":     portal.PortalRegion,
		"Status":     portal.PortalStatus,
		"PortalLink": itemUrl,
	}
	requestBody.SetAdditionalData(additionalData)
	return g.appClient.Sites().BySiteId(siteId).Lists().ByListId(listId).Items().ByListItemId(portal.ListID).Fields().Patch(context.Background(), requestBody, nil)
}
```

Would really appreciate your input for that matter!
Cheers,
Bernd

**Comment by baywet** [comment]
Hi @beschick 
Thanks for using the Go SDK and for reaching out.
Can you share the JSON payload you're sending? And the one you'd like to send?
Also, have you tried [using UntypedObject](https://github.com/microsoft/kiota-abstractions-go/blob/d5022cb21cecb8fdb576bede4d678fa68522532a/serialization/untyped_object.go#L9) instead of an interface map?

**Comment by ctxlabeu** [comment, marked as answer]
Thanks for your response @baywet!

I think I am not even struggling specifically with Go in that matter, but more with the right method to Patch a list item in SharePoint with GRAPH REST API in general.

The endpoint to get a list item's field is:
```
GET https://graph.microsoft.com/v1.0/sites/:siteid/lists/:listId/items/:itemId?expand=fields
```
The payload of an existing item for that specific column, looks like that. The response object is truncated for clarity.
```json
    "fields": {
        "Title": "GUEST WIFI access code hotspot 1",
        "Accesscode": "123456",
        "Region": "EMEA",
        "PortalLink": {
            "Description": "Portal 1",
            "Url": "https://guestportal1.company.com"
        },
        "Comments": "test",
        "id": "1",
        "ContentType": "Item",
       }

```

The equivalent REST API endpoint to UPDATE (PATCH) an SPO item is the following:
`PATCH https://graph.microsoft.com/v1.0/sites/:siteid/lists/:listId/items/:itemId/fields`

A working PATCH body would be this:
```json
{
    "Accesscode": "125669"
}
```
I was expecting to work it the same way with an URL field when patching it, but that does not work out obviously
```json
{
 "PortalLink": {
    "Url": "https://guestportal1A.company.com",
    "Description": "WIFI portal 1A"
  }
}
````
By trying this with REST, I recognize that even that payload does not work with JSON.
```
{
    "error": {
        "code": "invalidRequest",
        "message": "Invalid request"
    }
}
```

So, my summary is: I am not able to componse the proper PATCH payload as it seems :(

Reference: [https://learn.microsoft.com/en-us/graph/api/listitem-update?view=graph-rest-1.0&tabs=http](url)

**Reply by baywet** [reply]
Thanks for the additional information.
This repository is dedicated to the SDK itself, the service team is not watching it.
To get answers about the APIs specifically, please ask your question with the REST/JSON elements on Microsoft Q&A with the Microsoft Graph tag. Service team members are watching that forum.

**Reply by ctxlabeu** [reply]
Yeah, sure. Not an issue/question of the SDK itself. Generic Graph REST API topic.
